tests/transfer/storage/test_google_cloud.py

```diff
@@ -0,0 +1,45 @@
+"""Tests targeting GPC external storage
```

Comment:
I think this approach of mocking by patching *our own code* as opposed to mocking the GCP interaction in a lower level is not useful for real testing - it tests very little in reality. 

I suggest looking into one of the following approaches: 
1. Writing automated tests that *actually use GCP* if a token / some env var is provided, and then using `pytest-vcr` to record and mock network interactions so this can continue to run without real access to GCP. This has the benefit of working well and being quite generic, and the mocking is done on a *very low* level (even lower than the Google code). 

2. Monkey-patching the GCP client (`self.storage_client`) and not our own class, so that all the code in our own class actually runs. This would be easier with some sort of DI rather than what we do here, but can still be done easily given Python's flexible object model  (would be better if `__init__` wouldn't try to initialize the bucket though)